MapboxVision/Services/Recording/RecordCoordinator.swift

```diff
@@ -96,104 +101,130 @@ final class RecordCoordinator {
     }
 
     func stopRecording() {
-        guard isRecording, isReady else { return }
-
-        stopRecordingInBackgroundTask = UIApplication.shared.beginBackgroundTask()
-
         processingQueue.async { [weak self] in
-            guard let self = self else { return }
+            guard let self = self, self.isRecordingInternal, self.isReady else { return }
+
+            self.stopRecordingInBackgroundTask = UIApplication.shared.beginBackgroundTask()
```

Comment:
Could you please check that after moving this to async block it will still be called before suspending execution.
If you're testing it with Teaser please make sure to disable background mode "location updates".

Reply:
yep

Reply:
Have you got a chance to ensure it's working properly?

Reply:
yes, works as expected